### DISCUSSION TITLE
Can you share an example of API markdoc for example https://stripe.com/docs/api/charges/retrieve?

### DISCUSSION THREAD

**Discussion by kryltsov** [markdoc/markdoc#449] [Q&A]
I have checked the Markdoc site and actual code for it shared in a public repo but can not find a good example of a Markdoc page that can show a page like https://stripe.com/docs/api/charges/retrieve for example.

I understand that the code probably contains some logic you do not want to share/expose. Maybe it can be replaced with some dummy logic.

Thank you

**Comment by oasissoman** [comment]
OASISS BASINESS

**Comment by oasissoman** [comment]
oasissoman

**Comment by maaz-munir** [comment]
did you figure it out? @kryltsov

**Reply by kryltsov** [reply]
No :(

**Reply by robfish** [reply]
Would love to see the sample if one is shared !!

**Comment by mulan-stripe** [comment]
Hi! Can you share a bit more specifically what on that page you're interested in knowing about? You are correct in assuming there is a lot of Stripe specific code we are unable to share, but I may be able to help answer specific questions.

**Reply by go-try-it** [reply]
Totally understand that and BTW am loving Markdoc in general !

For me at least it was the component(s) that provided the in-situ swagger style format of the whole page, with embededded multi-language code boxes and response messages. Love the layout and was just curious about the components used.

TBH I'd love to see some kind of open-source component library - that'd really accelerate adoption and sharing/reuse !!!

**Reply by matv-stripe** [reply]
Hi @go-try-it ! We have built a system that takes the data from the API (similar to how we generate the Open API spec) and create Markdoc files for each resource. We also built custom React components on top of Stripe's design system that map to the Markdoc tags we output. For example, the Markdoc tags for the charges page might look something like:

![CleanShot 2024-06-25 at 13 14 19@2x](https://github.com/markdoc/markdoc/assets/55463370/1ce2b955-c86d-446d-b417-8d5e849f8844)


We do not have plans to open source anything at the moment. Here are resources to learn more about the code generation piece though!
* https://brandur.org/fragments/stripe-codegen
* https://www.thestrangeloop.com/2021/artisanal-machine-generated-api-libraries.html
